packages/cli/src/commands/setup/deploy/providers/coherence.js

```diff
@@ -0,0 +1,112 @@
+// import terminalLink from 'terminal-link'
+import fs from 'fs'
+import path from 'path'
+
+import { getSchema, getConfig } from '@prisma/internals'
+import { Listr } from 'listr2'
+
+import { errorTelemetry } from '@redwoodjs/telemetry'
+
+import { getPaths, writeFilesTask } from '../../../../lib'
+import c from '../../../../lib/colors'
+import { printSetupNotes, addFilesTask } from '../helpers'
+import {
+  COHERENCE_HEALTH_CHECK,
+  DATABASE_YAML,
+  COHERENCE_YAML,
+} from '../templates/coherence'
+
+export const command = 'coherence'
+export const description = 'Setup Coherence deploy'
+
+export const getCoherenceYamlContent = async () => {
+  if (!fs.existsSync('api/db/schema.prisma')) {
+    return {
+      path: path.join(getPaths().base, 'coherence.yml'),
+      content: COHERENCE_YAML(''),
+    }
+  } else {
+    const schema = await getSchema('api/db/schema.prisma');
+    const config = await getConfig({ datamodel: schema });
+    var detectedDatabase:string = config.datasources[0].activeProvider;
+
+    if ((detectedDatabase === 'mysql') ||  (detectedDatabase === 'postgresql')) {
+      if (detectedDatabase === 'postgresql') {detectedDatabase = 'postgres'}
+      return {
+        path: path.join(getPaths().base, 'coherence.yml'),
+        content: COHERENCE_YAML(DATABASE_YAML(detectedDatabase)),
+      }
+    } else {
+      printSetupNotes('Only mysql & postgresql prisma DBs are supported on Coherence at this time...')
+      return {
+        path: path.join(getPaths().base, 'coherence.yml'),
+        content: COHERENCE_YAML(''),
+      }
+    }
+  }
+}
+
+// any notes to print out when the job is done
+const notes = [
+  'You are ready to deploy to Coherence!\n',
+  'Go to https://app.withcoherence.com to create your account and setup your cloud/github connections.',
+  'Check out the deployment docs at https://docs.withcoherence.com for detailed instructions and more information.\n',
+  'Reach out to redwood@withcoherence.com with any questions! We are here to support you...'
+]
+
+
+const additionalFiles = [
+  {
+    path: path.join(getPaths().base, 'api/src/functions/health.js'),
+    content: COHERENCE_HEALTH_CHECK,
+  },
+]
```

Comment:
We're going to need to make the file extension respect the project type; you can import `isTypeScriptProject` from `@redwoodjs/cli-helpers`:

```suggestion
const additionalFiles = [
  {
    path: path.join(
      getPaths().api.functions,
      `health.${isTypeScriptProject ? 'ts' : 'js'}`
    ),
    content: COHERENCE_HEALTH_CHECK,
  },
]]
```